lib/ui/send/interactor/send_state.dart

```diff
@@ -7,8 +7,19 @@ class SendState with _$SendState {
   const factory SendState({
     @Default(PageState.initial) PageState pageState,
     PageCommand? command,
-    @Default('0') String userEnteredAmount,
+    @Default(null) String? userEnteredAmount,
+    @Default(null) String? memo,
     required UserProfileData receiverUser,
     required WalletTokenData tokenData,
+    @Default(false) bool isSendButtonLoading,
   }) = _SendState;
+
+  bool get isSubmitEnabled {
+    final parsedAmount = double.tryParse(userEnteredAmount ?? '0') ?? 0;
+    return parsedAmount > 0 && parsedAmount <= (tokenData.userOwnedAmount ?? 0);

```

Comment:
why can userOwnedAmount be null and if it is null - I assume it means unknown - why do we treat it like 0?